### PR TITLE
Add: support 1M context in DeepSeek V4 HCA decode

### PR DESCRIPTION
**Pull request by sjduan**
- raise the HCA decode context ceiling to 1M without moving the global
  Flash model ceiling
- pass token-local QKV RoPE rows as `[T_DYN, ROPE_HEAD_DIM]` and
  compressed-boundary rows as a separate `[B, ROPE_HEAD_DIM // 2]`
  argument, so no 1M RoPE table is materialized on host or device
- drop the `cmp_sparse_indices` argument; `sparse_attn_hca` now takes
  `position_ids` and `kv_seq_lens` and derives each token's visible
  compressed row count itself
- size compressed work from the runtime `cmp_block_table` width as
  page-aligned 128-row items; work past a token's visible rows stores
  neutral softmax partials and skips its QK/PV matmuls
- split raw gather/transpose, compressed work, streaming merge, output
  projection, and HC-post lifetimes with manual scopes
- separate the per-request context ceiling `HCA_MAX_COMPRESSED_ROWS`
  from the batch-shared pool capacity `HCA_COMPRESSED_POOL_ROWS`
- zero-fill every compressed work page before its block-table lookup so
  an unmapped page cannot feed unwritten rows to the softmax
- compare every lane of a contiguous gather run instead of its two
  endpoints, so an interior unmapped slot takes the masked path
- share one token-count check between the TP1 and distributed fixtures
  and reject a request count outside `[1, B]`
- retain the `--tp` routing and inlined distributed output structure
  from #1003
- validated on A2/A3 at TP1, TP2, and TP4 for 128, 16K, and 1M contexts
  and the mixed `(16K, 16K, 128K, 512K)` fixture

The ratio-128 path has no indexer, so its compressed tail is the
deterministic full cache and the old Top-K list was always an identity
permutation whose length scaled with `MAX_SEQ_LEN`. Deriving the row
count from `position_ids` and `kv_seq_lens` turns that compile-time
bound into a runtime one and lets short requests skip matmuls a data
mask could not. Related to #962.

### CODE REVIEW

File: models/deepseek_v4_flash_dspark/decode_hca.py [hunk 1]
```diff
@@ -878,24 +793,140 @@ def golden_decode_hca(tensors):
...
-    shared_freqs_cos, shared_freqs_sin = build_rope_tables(M, COMPRESS_RATIO, dtype=torch.bfloat16)
+    starts = _hca_start_positions(start_pos, batch=batch)
+    positions = position_ids_from_starts(starts, seq=S).contiguous()
+    token_freqs_cos, token_freqs_sin = _hca_token_rope_tables(positions)
+    boundary_positions = starts.to(torch.int64) - starts.to(torch.int64).remainder(COMPRESS_RATIO)
+    boundary_cos, boundary_sin = _hca_token_rope_tables(boundary_positions)
+    cmp_freqs_cos = torch.zeros(B, ROPE_HEAD_DIM // 2, dtype=torch.float32)
+    cmp_freqs_sin = torch.zeros(B, ROPE_HEAD_DIM // 2, dtype=torch.float32)
+    cmp_freqs_cos[:batch] = boundary_cos[:, :ROPE_HEAD_DIM // 2].float()
+    cmp_freqs_sin[:batch] = boundary_sin[:, :ROPE_HEAD_DIM // 2].float()
```
[the remark this answers is not in the document]
Comment (sjduan) [reply models/deepseek_v4_flash_dspark/decode_hca.py:939]: current format is for accuracy ensure.
